src/palace/manager/api/admin/config.py

```diff
@@ -24,38 +26,20 @@ class AdminClientFeatureFlags(ServiceConfigurationWithLimitedEnvOverride):
         description="Show QuickSight dashboards only for sysadmins.",
     )
 
-    # The following fields CANNOT be overridden by environment variables.
-    # Setting `const=True` ensures that the default value is not overridden.
-    # Add them to the one of the `environment_override_*` Config settings
-    # below to prevent them from being overridden.
-    # NB: Overriding the `env_prefix` with `env=...` here may lead to
-    #     incorrect values in warnings and exceptions, since `env_prefix`
-    #     is used to generate the full environment variable name.
-    enable_auto_list: bool = Field(
+    # The following fields CANNOT be overridden by environment variables, so
+    # their types are set to Literal[True] to prevent them from being overridden.
+    enable_auto_list: Literal[True] = Field(
```

Comment:
I think this will behave more like an `environment_override_error_fields` (which we're not currently using) field, which would probably prevent the admin UI working.

On the contrary, the way `environment_override_[warning]_fields` (which we are using) behaves is that it:
- prevents a change to the variable; and
- emits a warning, so that we might notice that something was being set that should not be set, so that we could go clean it up.

I might be missing it, but I didn't see a test for this case.

Reply:
How would you feel about removing `Literal[True]` as I have here, and just adding a todo comment to go in and remove these feature flags if we no longer need them?

I implemented this in https://github.com/ThePalaceProject/circulation/pull/2080/commits/d890d47937b905fc01abeb00b7ee1c35552a77be.

Reply:
I think that would be better than delayed failing. Tradeoffs. 🤷🏽‍♂️